### ISSUE TITLE
Accidential removal of objects, when import source is on error state

### ISSUE THREAD

**Issue by chgarling** [Icinga/icingaweb2-module-director#2397]
## Expected Behavior
Objects like hosts or services should not be removed, when the import source like Foreman is not available, for example due to a 503 error.

## Current Behavior
Our Foreman was not available with an 503 error because it was crashed. We are using the Foreman REST API as datasource for the hosts. Since the REST API was not available the Icinga2 director removed all hosts, which were imported from this data source earlier.

## Possible Solution
Put some magic in the Icinga2 director to prevent this behavior.

## Steps to Reproduce (for bugs)
Use Foreman REST API as datasource for host object creation. Then serve a 503 error instead a valid API response.

## Your Environment
* Director version (System - About): 1.8.0
* Icinga Web 2 version and modules (System - About): 2.8.2
* Icinga 2 version (`icinga2 --version`): r2.12.4-1
* Operating System and version: Debian 10.10
* Webserver, PHP versions: Apache2 2.4.38-3+deb10u5 / PHP 7.3.29-1~deb10u1

**Comment by EmmaTinten** [comment]
We have several import sources definied which are importing hosts / services of different APIs. Some of the APIs are in, let's say an unstable state. Sometimes they are reachable but not providing the neccessary import data due to missing backends. So the importer retrieves empty lists. The API itself is reachable but provides no data.

**It would be great to add an option to the automation process of the Icinga Director which denies the synchronization if the import source deliveres empty results (sets the counter for imported lines to 0) .** 

At the moment we're using simple import scripts which are writing files. This script will not overwrite such a file if the resulting file will be empty (still, we can force this to really set the imported lines to zero). This works but it forces us to always create such an import script with the corresponding trigger (e.g. systemd timers).

The APIs are imported frequently and that's not possible to change. Sometimes, imaging changes on the backends, misconfigured firewalls and so on, the import is triggerd when the api will only deliver empty results. The next run several minutes later will receive all data. 

It would be great to have  simple boolean value which denies the import of an import source if the result will remove all existing lines.

**Comment by Thomas-Gelf** [comment]
Which ImportSource type are you using?

**Comment by EmmaTinten** [comment]
This occurs when using REST APIs directly. But this is no error in the handling of the director - the APIs sometimes deliver empty results (but the correct structure).

**Comment by EmmaTinten** [comment]
Yeah great, @Thomas-Gelf . I´ll open a dedicated request. Thank you!

**Comment by Thomas-Gelf** [comment]
You're welcome!

**Comment by chgarling** [comment]
Hi Thomas,

sorry for the late reply. We lost focus on this issue because of other projects, but I still have an internal ticket to go on with this.

I have to find a way we can reproduce the scenario. We don't have any Foreman and Icinga2 test systems, and I don't want to try and error in production.

I will come back to you as soon as possible.

Regards, Christian

**Comment by Thomas-Gelf** [comment]
Thank you!

**Comment by Thomas-Gelf** [comment]
@chgarling: in the meantime, what ImportSource type are you using? Is it really the RestApi Import shipped with the Director? If unsure, Screenshot of it's configuration could help.

**Comment by chgarling** [comment]
Source type is REST API.

**Comment by Tqnsls** [comment]
any update on this @chgarling ?
